mempool/mempool_test.go

```diff
@@ -163,6 +163,18 @@ func TestMempoolFilters(t *testing.T) {
 	}
 }
 
+func TestMempoolUpdate(t *testing.T) {
```

Comment:
Perhaps rename to `TestMempoolUpdate_cache`, or something to that extent. From the name, I thought it was going to test that update removed the tx from the mempool list itself.